lib/services/messenger.ts

```diff
@@ -64,8 +64,10 @@ export class MessengerService extends ServiceScaffold<string> implements Service
 					listenerMethod: (_registration: ServiceRegistration, event: ServiceScaffoldEvent): Promise<void> => {
 						// This translates and enqueues any events that match a registered interest.
 						if (_.includes(this.eventsRegistered, subTranslator.eventIntoMessageType(event))) {
-							return subTranslator.eventIntoMessage(event)
-							.then(this.queueData);
+							return subTranslator.eventIntoMessages(event)
+							.then((messages) => {
+								_.forEach(messages, this.queueData);
```

Comment:
Does typescript automatically return the last line? Otherwise, this may not have completed before it's used.

Reply:
This (correct me if I'm wrong Andrew) should end up queuing an event, which doesn't wait for a response but pushes the data into an async event queue. So, it doesn't actually need to wait for completion.

Reply:
This (correct me if I'm wrong Andrew) should end up queuing an event, which doesn't wait for a response but pushes the data into an async event queue. So, it doesn't actually need to wait for completion.

Reply:
No correction needed, Hedss is entirely correct.